util/topsql/tracecpu/profile.go

```diff
@@ -276,16 +301,24 @@ func (sp *sqlCPUProfiler) hasExportProfileTask() bool {
 	return has
 }
 
-// IsEnabled return true if it is(should be) enabled. It exports for tests.
-func (sp *sqlCPUProfiler) IsEnabled() bool {
-	return variable.TopSQLEnabled() || sp.hasExportProfileTask()
+// ShouldProfile return true if it's required to profile. It exports for tests.
+func (sp *sqlCPUProfiler) ShouldProfile() bool {
```

Comment:
Why it is not checking the enabling state?

Reply:
Which enabling state?